config/config.go

```diff
@@ -179,3 +188,18 @@ func GetOptionalParameters(config *Config, defaultLidarDataRateMsec, defaultIMUD
 
 	return optionalConfigParams, nil
 }
+
+func validateModes(
+	optionalConfigParams OptionalConfigParams,
+	imuEnabled bool,
+) error {
+	offlineMode := optionalConfigParams.LidarDataRateMsec == 0
+	localizationMode := !optionalConfigParams.EnableMapping
+	if localizationMode && offlineMode {
+		if imuEnabled {
+			return errLocalizationInOfflineModeIMU
+		}
+		return errLocalizationInOfflineMode
```

Comment:
i'm also confused why we return errLocalizationInOfflineMode that mentions data_rate_msec in its message when we aren't checking data_rate_msec at all?

Reply:
discussed offline

Reply:
when IMU is not enabled, `optionalConfigParams.LidarDataRateMsec` comes from `data_rate_msec`